instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/internal/instrumentation/runtimemetrics/RuntimeMetricsAutoConfiguration.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.spring.autoconfigure.internal.instrumentation.runtimemetrics;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.spring.autoconfigure.internal.ConditionalOnEnabledInstrumentation;
+import io.opentelemetry.instrumentation.spring.autoconfigure.internal.properties.ConfigPropertiesBridge;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import java.util.Comparator;
+import java.util.Optional;
+import javax.annotation.PreDestroy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.event.EventListener;
+
+/**
+ * Configures runtime metrics collection.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
+@ConditionalOnEnabledInstrumentation(module = "runtime-telemetry")
+@Configuration
+public class RuntimeMetricsAutoConfiguration {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(RuntimeMetricsAutoConfiguration.class);
+
+  private Runnable shutdownHook;
+
+  @PreDestroy
+  public void stopMetrics() {
+    if (shutdownHook != null) {
+      shutdownHook.run();
+    }
+  }
+
+  @EventListener
+  public void handleApplicationReadyEvent(ApplicationReadyEvent event) {
+    ConfigurableApplicationContext applicationContext = event.getApplicationContext();
+    OpenTelemetry openTelemetry = applicationContext.getBean(OpenTelemetry.class);
+    ConfigPropertiesBridge config =
+        new ConfigPropertiesBridge(applicationContext.getBean(ConfigProperties.class));
+
+    double version = javaVersion();
+    Optional<RuntimeMetricsProvider> metricsProvider =
+        applicationContext.getBeanProvider(RuntimeMetricsProvider.class).stream()
+            .sorted(Comparator.comparing(RuntimeMetricsProvider::minJavaVersion).reversed())
+            .filter(provider -> provider.minJavaVersion() <= version)
+            .findFirst();
+
+    if (metricsProvider.isPresent()) {
+      metricsProvider.get().start(openTelemetry, runnable -> shutdownHook = runnable, config);
+    } else {
+      logger.debug("No runtime metrics instrumentation available for Java {}", version);
+    }
+  }
+
+  private static int javaVersion() {
+    try {
+      return Integer.parseInt(System.getProperty("java.specification.version"));
+    } catch (NumberFormatException nfe) {
+      // The value of the java.specification.version property is 1.8 with a Java 8 runtime.
```

Comment:
elsewhere we use `Double.parseDouble` to not be affected by this

Reply:
changed as a result of https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/13173#discussion_r1946918008 - happy to change it back (`1.8` is a valid double after all)

Reply:
done